contracts/interchain-token-service/tests/deploy_remote_canonical_token.rs

```diff
@@ -62,9 +62,43 @@ fn deploy_remote_canonical_token_succeeds() {
     }
     .abi_encode(&env);
 
-    let deployed_token_id = client
-        .mock_all_auths_allowing_non_root_auth()
-        .deploy_remote_canonical_token(&token_address, &destination_chain, &spender, &gas_token);
+    let transfer_token_auth = mock_auth!(
+        env,
+        spender,
+        gas_token.transfer(spender, gas_service.address, gas_token.amount),
+        &[]
+    );
+
+    let pay_gas_auth = mock_auth!(
+        env,
+        spender,
+        gas_service.pay_gas(
+            client.address,
+            its_hub_chain,
+            its_hub_address,
+            payload,
+            spender,
+            gas_token,
+            Bytes::new(&env)
+        ),
+        &[(transfer_token_auth.invoke).clone()]
```

Comment:
```suggestion
        &[transfer_token_auth.invoke]
```
it's not being used later so is a clone needed?

Reply:
I tried the same, but the compiler gives this error. `&[transfer_token_auth.invoke]` would not work, we should use `.clone()` here.
```
error[E0308]: mismatched types
  --> contracts/interchain-token-service/tests/deploy_remote_canonical_token.rs:83:11
   |
83 |         &[transfer_token_auth.invoke]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `MockAuthInvoke<'_>`, found `&MockAuthInvoke<'_>`

For more information about this error, try `rustc --explain E0308`.
```